core/src/main/java/com/linecorp/armeria/common/RequestContextAwareEventLoop.java

```diff
@@ -48,7 +48,7 @@ protected EventLoop delegate() {
 
     @Override
     public EventLoop next() {
-        return delegate().next();
+        return new RequestContextAwareEventLoop(context(), delegate().next());
```

Comment:
Just returning `this` should be OK here. `EventLoop.next()` by definition returns itself. It's only useful when used in the context of `EventLoopGroup.next()`. :-)